src/test/java/com/opencredo/connect/venafi/tpp/log/EventLogSourceTaskTest.java

```diff
@@ -228,17 +238,33 @@ public void as_a_client_I_want_to_paginate_logs() {
 
         List<SourceRecord> page_1_of_logs = when_the_task_is_polled(task);
         then_the_logs_are_of_size(page_1_of_logs, 5);
+
+        then_the_logs_will_have_some_records_with_some_offset(page_1_of_logs, 5, 1L);
         //get next page
         List<SourceRecord> page_2_of_logs = when_the_task_is_polled(task);
         then_the_logs_are_of_size(page_2_of_logs, 5);
         then_the_number_of_logs_with_timestamp_is(5, page_2_of_logs, getTodayPlus(5));
+        then_the_logs_will_have_some_records_with_some_offset(page_2_of_logs, 1, 2L);
+        then_the_logs_will_have_some_records_with_some_offset(page_2_of_logs, 1, 3L);
+        then_the_logs_will_have_some_records_with_some_offset(page_2_of_logs, 1, 4L);
+        then_the_logs_will_have_some_records_with_some_offset(page_2_of_logs, 1, 5L);
+        then_the_logs_will_have_some_records_with_some_offset(page_2_of_logs, 1, 6L);
         //get next Page
         List<SourceRecord> page_3_of_logs = when_the_task_is_polled(task);
         then_the_logs_are_of_size(page_3_of_logs, 4);
         then_the_number_of_logs_with_timestamp_is(2, page_3_of_logs, getTodayPlus(6));
+        then_the_logs_will_have_some_records_with_some_offset(page_3_of_logs, 1, 7L);
+        then_the_logs_will_have_some_records_with_some_offset(page_3_of_logs, 2, 1L);
+        then_the_logs_will_have_some_records_with_some_offset(page_3_of_logs, 1, 2L);
 
         List<SourceRecord> page_4_of_logs = when_the_task_is_polled(task);
         then_the_logs_are_of_size(page_4_of_logs, 2);
+        then_the_logs_will_have_some_records_with_some_offset(page_4_of_logs, 2, 1L);
+
+    }
+
+    private void then_the_logs_will_have_some_records_with_some_offset(List<SourceRecord> page_1_of_logs, int someRecords, long someOffset) {
```

Comment:
I'm having difficulty understanding what you are testing here - maybe because it's out of context?
- What's "some"? 
- Why is it page **1** of the logs?
- Aren't `someRecords` actually the expectedNumberOfRecords?
- Isn't `someOffset` some king of expected offset? (What kind?)
- What are you actually counting down there? On a page of logs, get all log entries whose offset is the expected offset, and count how many there are? I don't recall what the data structure of the logs looks like, but is there really more than one with the same offset?

Maybe add a dummy payload somewhere in a docs or resources file so it's easier to reason about this?

Reply:
Okay, so, offset is Kafka's... still confused. Does that sourceOffset call return the date, then?

Reply:
So I should probably rename some things.
Page 1 of logs is wrong I used intellij to handle the method creation; It's a list of logs so it should probably be renamed as logs.
The someRecords are indeed I expect an amount of records with someApiOffset. Also be aware that if each record has the same date then the apiOffset will be incremented between each record.
I'll rename some things and see if it makes it easier to understand.